src/mono/wasm/Wasm.Build.Tests/WasmTemplateTests.cs

```diff
@@ -55,8 +55,7 @@ private void UpdateConsoleMainJs()
             string mainJsContent = File.ReadAllText(mainJsPath);
 
             mainJsContent = mainJsContent
-                .Replace(".create()", ".withConsoleForwarding().create()")
-                .Replace("[\"dotnet\", \"is\", \"great!\"]", "(await import(/* webpackIgnore: true */\"process\")).argv.slice(2)");
+                .Replace(".create()", ".withConsoleForwarding().create()");
```

Comment:
This needed? It won't get automatically triggered by runArgs having forward console set?

Reply:
Patching it means that we aren't really testing the templates as a user would have them.

Reply:
there is no `runArgs` for production code, just for `test-main.js`

Reply:
:/ I thought we were supporting it to allow WasmAppHost to use this :/

Reply:
> Patching it means that we aren't really testing the templates as a user would have them.

We could not have both, A) console forwarding and B) untouched template

Reply:
We could, with runArgs. We have this builder pattern, which could maybe do something similar to what aspnetcore does for adding arguments from environment variables, command line, or other sources like json files.

Reply:
Why is console forwarding needed for console template test anyway?

Reply:
re:run-args, we can discuss in a separate issue.

Reply:
We need proper design for all those things first
https://github.com/dotnet/runtime/issues/70892

Reply:
We should not be taking over the application argument with magic like `--forward-console`

Reply:
that's not an application argument though. It's an argument for the host, which then we need to communicate as a setting/config to the runtime.

Reply:
> Why is console forwarding needed for console template test anyway?

I thought we assert the output in the test.

Reply:
Oh, not in the console, we don't need it there.

Reply:
It would be running with node, and that should show up in stdout, no?

Reply:
> that's not an application argument though. It's an argument for the host, which then we need to communicate as a setting/config to the runtime.

![image](https://user-images.githubusercontent.com/271576/192537718-2a852d21-ddce-4a08-bc34-fa44f3780a82.png)

Are we able to add the arguments in the same way as the existing args. That is before the app name ?
Do we handle `--` correctly ?

Reply:
That doesn't always work correctly, yet - *experimental*. But doesn't change the fact that it will be (wasm)host argument, and not an application one.

Reply:
underlying node.exe also has it's own runtime vs application arguments separated by `--`.

Not all the dotnet apps in node need to be executed by `dotnet run` host. 

I guess my concern is that our mono runtime arguments should not invade what node would have after `--`.
I consider the dotnet app being first class citizen in nodejs. Not plugin of the host app.

Reply:
We can deal with this in the design stuff.